src/pip/_internal/vcs/bazaar.py

```diff
@@ -6,8 +6,15 @@
 from pip._vendor.six.moves.urllib import parse as urllib_parse
 
 from pip._internal.utils.misc import display_path, path_to_url, rmtree
+from pip._internal.utils.typing import MYPY_CHECK_RUNNING
 from pip._internal.vcs.versioncontrol import VersionControl, vcs
 
+if MYPY_CHECK_RUNNING:
+    from typing import Optional, Tuple
+    from pip._internal.vcs.versioncontrol import RevOptions
+
+    AuthInfo = Tuple[Optional[str], Optional[str]]
+
```

Comment:
@chrahunt told me he prefers two empty lines before the first line of code. (This suggestion applies to the other files as well.)